### PR TITLE
Revert ODATA-1416 after ODATA-1420 is RESOLVED

### PR DESCRIPTION
**Pull request by HeikoTheissen**
CHANGELOG:
- Let the terms `Capabilities.(Insert,Update,Delete)Restrictions` no longer apply to `EntityType` and remove tag `Core.AppliesToTypeIfDynamic`
- Delete term `Core.AppliesToTypeIfDynamic`, since no longer needed
- Let all capabilities terms plus `NavigationRestrictions` apply to arbitrary container-bound collections.

### CODE REVIEW

**Comment by HeikoTheissen** [comment]
TC 2021-09-30: Neither the old nor the new version allows annotating `InsertRestrictions` for subitems if the path expression consists of the `insertable` property on the item.

**Comment by HeikoTheissen** [comment]
> TC 2021-09-30: Neither the old nor the new version allows annotating `InsertRestrictions` for subitems if the path expression consists of the `insertable` property on the item.

https://github.com/oasis-tcs/odata-vocabularies/blob/098a7a1633158329e908bcec217a245f1acd8025/vocabularies/Org.OData.Capabilities.V1.xml#L602-L605

allows the following annotation
```json
"HeadersEntitySet": {
  "@Capabilities.InsertRestrictions": {
    "NavigationProperty": "items",
    "Insertable": {"$Path": "HeaderChangeable"},
    "NonInsertableProperties": ["items/ItemUUID"]
  }
}
```

**Review by ralfhandl — DISMISSED** [review]

**Comment by HeikoTheissen** [comment]
Shall we include an example for the `InsertRestrictions`?

**Comment by ralfhandl** [comment]
> Shall we include an example for the `InsertRestrictions`?

Yes, please

**Review by ralfhandl — APPROVED** [review]

File: examples/Org.OData.Capabilities.V1.capabilities.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      "NavigationProperty": "Items",
+      "InsertRestrictions": {
+     // "NonInsertableProperties": ["Items/uuid"],
+        "Insertable": {"$Path": "canInsertItems"}
+      },
+      "UpdateRestrictions": {
+     // "Updatable": {"$Path": "Items/canUpdate"},
+     // "NonUpdatableProperties": ["Items/uuid"],
+        "FilterSegmentSupported": {"$Path": "canUpdateSubsetOfItems"}
+      }
```
Comment (HeikoTheissen) [inline examples/Org.OData.Capabilities.V1.capabilities.md:37]: This shows that the decision where to annotate something is made per property (such as `UpdateRestrictions/Updatable`), not per term (such as `UpdateRestrictions`).

**Review by ralfhandl — APPROVED** [review]

**Pull request merged**